packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts

```diff
@@ -0,0 +1,271 @@
+    const stack = new Stack(app);
+    const shareAccountId = '012345678901';
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    portfolio.share(shareAccountId, {
+      shareTagOptions: true,
+      acceptLanguage: AcceptLanguage.EN,
+    });
+
+    expect(stack).toHaveResource('AWS::ServiceCatalog::PortfolioShare', {
+      AccountId: shareAccountId,
+      ShareTagOptions: true,
+      AcceptLanguage: 'en',
+    });
+  }),
+
+  test('portfolio share without share tagOptions', () => {
+    const app = new App();
+    const stack = new Stack(app);
+    const shareAccountId = '012345678901';
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    portfolio.share(shareAccountId, { shareTagOptions: false });
+
+    expect(stack).toHaveResource('AWS::ServiceCatalog::PortfolioShare', {
+      AccountId: shareAccountId,
+      ShareTagOptions: false,
+    });
+  }),
+
+  test('portfolio share without explicit share tagOptions', () => {
+    const app = new App();
+    const stack = new Stack(app);
+    const shareAccountId = '012345678901';
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    portfolio.share(shareAccountId);
+
+    expect(stack).toHaveResource('AWS::ServiceCatalog::PortfolioShare', {
+      AccountId: shareAccountId,
+    });
+  }),
+
+  test('portfolio principal association with role type', () => {
+    const app = new App();
+    const stack = new Stack(app);
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    const role = new Role(stack, 'TestRole', {
+      assumedBy: new AccountRootPrincipal(),
+    });
+
+    portfolio.giveAccessToRole(role);
+
+    expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioPrincipalAssociation', {});
+  }),
+
+  test('portfolio principal association with user type', () => {
+    const app = new App();
+    const stack = new Stack(app);
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    const user = new User(stack, 'TestUser', {});
+
+    portfolio.giveAccessToUser(user);
+
+    expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioPrincipalAssociation', {});
+  }),
+
+  test('portfolio principal association with group type', () => {
+    const app = new App();
+    const stack = new Stack(app);
+
+    const portfolio = new Portfolio(stack, 'MyPortfolio', {
+      name: 'testPortfolio',
+      provider: 'testProvider',
+    });
+
+    const group = new Group(stack, 'TestGroup', {});
+
+    portfolio.giveAccessToGroup(group);
+
+    expect(stack).toHaveResourceLike('AWS::ServiceCatalog::PortfolioPrincipalAssociation', {});
+  }),
```

Comment:
Can you please group these tests? If I counted correctly, 7 tests in a row all start with:

```ts
    const app = new App();
    const stack = new Stack(app);

    const portfolio = new Portfolio(stack, 'MyPortfolio', {
      name: 'testPortfolio',
      provider: 'testProvider',
    });
```

This really should be extracted to a `describe()` block, and put in a `beforeEach()` helper, to avoid all of this duplication.

Reply:
yes reorganized into describe blocks that can use `beforeEach()` for to make more DRY